village/src/components/Smurf.js

```diff
@@ -1,13 +1,51 @@
 import React from 'react';
+import EditSmurf from './EditSmurf'
 
-const Smurf = props => {
-  return (
-    <div className="Smurf">
-      <h3>{props.name}</h3>
-      <strong>{props.height} tall</strong>
-      <p>{props.age} smurf years old</p>
-    </div>
-  );
+class Smurf extends React.Component{
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      edit: false
+    }
+  }
+
+  edit = (edit) => {
+    let id = this.props.id
+    this.props.edit(id, edit)
+    this.setState({edit: false});
+  }
+
+  clickHandler = () => {
+    if (this.state.edit) {
```

Comment:
If you are toggling something in state you can do this in one line
`this.setState({ this.state.whatever: !this.state.whatever })`
So in case 
`this.setState({ this.state.edit: !this.state.edit })`